internal/controller/stages/stages.go

```diff
@@ -974,6 +988,32 @@ func (r *reconciler) clearApprovals(
 	return nil
 }
 
+func (r *reconciler) clearAnalysisRuns(
+	ctx context.Context,
+	stage *kargoapi.Stage,
+) error {
+	if r.rolloutsClient == nil {
+		return nil
+	}
+
+	namespace := r.getAnalysisRunNamespace(stage)
+	if err := r.rolloutsClient.DeleteAllOf(
+		ctx,
+		&rollouts.AnalysisRun{},
+		client.InNamespace(namespace),
+		client.MatchingLabels(map[string]string{
+			kargoapi.StageLabelKey: stage.Name,
```

Comment:
Ha. I forgot we labeled the ARs this way. This is certainly convenient when it's time to delete these.